src/viewer.js

```diff
@@ -446,23 +495,608 @@ class VolumeImageViewer {
+
+        const path = channel.pyramidFrameMappings[z][index];
+        if (path === undefined) {
+          console.warn("tile " + index + " not found at level " + z);
+          return (null);
+        }
+        let url = options.client.wadoURL +
+          "/studies/" + channel.pyramidMetadata[z].StudyInstanceUID +
+          "/series/" + channel.pyramidMetadata[z].SeriesInstanceUID +
+          '/instances/' + path;
+        if (options.retrieveRendered) {
+          url = url + '/rendered';
+        }
+        return (url);
+      }
+      /*
+       * Define custom tile loader function, which is required because the
+       * WADO-RS response message has content type "multipart/related".
+      */
+      const tileLoadFunction = async (tile, src) => {
+        const img = tile.getImage();
+
+        const z = tile.tileCoord[0];
+        const columns = channel.pyramidMetadata[z].Columns;
+        const rows = channel.pyramidMetadata[z].Rows;
+        const samplesPerPixel = channel.pyramidMetadata[z].SamplesPerPixel; // number of colors for pixel
+        const BitsAllocated = channel.pyramidMetadata[z].BitsAllocated; // memory for pixel
+        const PixelRepresentation = channel.pyramidMetadata[z].PixelRepresentation; // 0 unsigned, 1 signed
+
+        const { contrastLimitsRange, color, opacity } = channel;
+
+        if (src !== null) {
+          const studyInstanceUID = DICOMwebClient.utils.getStudyInstanceUIDFromUri(src);
+          const seriesInstanceUID = DICOMwebClient.utils.getSeriesInstanceUIDFromUri(src);
+          const sopInstanceUID = DICOMwebClient.utils.getSOPInstanceUIDFromUri(src);
+          const frameNumbers = DICOMwebClient.utils.getFrameNumbersFromUri(src);
+        
+          if (options.retrieveRendered) {
+            // Figure out from the metadata if this is a color image dataset
+            // if it is, use png mediatype and png transfer syntax to just get pngs
+            // Otherwise, get the data as an octet-stream and use that
+            // TO DO: should we get always png and decompress for monochorme channels (i.e. samplesPerPixel === 1)?
+            // https://github.com/cornerstonejs/codecs
+  
+            let mediaType = 'application/octet-stream';
+            let transferSyntaxUID = '1.2.840.10008.1.2.1';
+            if (samplesPerPixel !== 1) {
+              mediaType = 'image/png';
+              transferSyntaxUID = '';
+            }
+            const retrieveOptions = {
+              studyInstanceUID,
+              seriesInstanceUID,
+              sopInstanceUID,
+              frameNumbers,
+              mediaTypes: [
+                { mediaType, transferSyntaxUID }
+              ]
+            };
+            if (options.includeIccProfile) {
+              retrieveOptions['queryParams'] = {
+                iccprofile: 'yes'
+              }
+            }
+            if (samplesPerPixel === 1) {
+              options.client.retrieveInstanceFrames(retrieveOptions).then(
```

Comment:
uhm that is true, but the if the viewer has been initialized with retrieveRendered = true, in the case of channel I still need to get the data uncompressed to make blending. Also the difference before wa just that in true, it was retrievening png, if false jpeg. Now it is the same for RGB images, but for channel it actaully always retrieve as octet-stream. This can change with point (5) of the PR. So probably it would make sense to discuss this again later on.

Reply:
Update, I have differenziated between channel and normal RGB images in fffb897, so there is not anymore this problem